ase/api/src/buffer.c

```diff
@@ -532,3 +533,40 @@ fpga_result __FPGA_API__ ase_fpgaGetIOAddress(fpga_handle handle, uint64_t wsid,
 	}
 	return result;
 }
+
+/*
+ * A single PASID is associated with the process, independent of the number
+ * of FPGA ports connected or the open/close sequence.
+ */
+static int process_pasid;
+
+fpga_result __FPGA_API__ ase_fpgaBindSVA(fpga_handle handle, uint32_t *pasid)
+{
+	struct _fpga_handle *_handle = (struct _fpga_handle *)handle;
+	fpga_result result = FPGA_OK;
+	int err;
+
+	if (!pasid)
+		return FPGA_INVALID_PARAM;
+
+	result = handle_check_and_lock(_handle);
+	if (result)
+		return result;
+
+	/* Allocate a PASID if the process doesn't have one yet */
+	if (!process_pasid) {
+		/* PASID is 20 bits, non-zero */
+		process_pasid = rand() & 0xfffff;
```

Comment:
Do you need something like 1 + rand() % 0xffffff so that it is non-zero?

Reply:
See line 561

Reply:
doh, I see.